### PR TITLE
Fix/merkle package benchmark test

### PR DESCRIPTION
**Pull request by vjdmhd**

### CODE REVIEW

File: merkle/common_test.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ func setupProofTestCases() ([]ProofTestCases, error) {
 	var merkleProofCases []ProofTestCases
 	for i := 0; i < len(maxCase); i++ {
 		var leavesHashes [][]byte
-		var Leaves []types.Leaf
+		var Leaves Leaves
```
Comment (blasrodri) [inline merkle/common_test.go:44]:
```suggestion
		var leaves Leaves
```

File: merkle/common_test.go [hunk 2]
```diff
@@ -74,19 +73,19 @@ func setupProofTestCases() ([]ProofTestCases, error) {
...
 }
 
-func combinations(leaves []types.Leaf) [][]types.Leaf {
-	return combine([]types.Leaf{}, leaves, [][]types.Leaf{})
+func combinations(leaves Leaves) Layers {
+	return combine(Leaves{}, leaves, Layers{})
 }
 
-func combine(active []types.Leaf, rest []types.Leaf, combinations [][]types.Leaf) [][]types.Leaf {
+func combine(active Leaves, rest Leaves, combinations Layers) Layers {
```
Comment (blasrodri) [inline merkle/common_test.go:80]: can you please document what does `combine` do?

File: merkle/index.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package merkle
+
+import "sort"
+
+// siblingIndecies returns indecirs of sibling elements
```
Comment (blasrodri) [inline merkle/index.go:5]:
```suggestion
// siblingIndecies returns indecies of sibling elements
```

File: merkle/index.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+func siblingIndecies(idxs []uint64) []uint64 {
```
Comment (blasrodri) [inline merkle/index.go:6]: maybe call it `mapSiblingIndex`?

File: merkle/index.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+// parentIndecies returns indecies of parent elements
+func parentIndecies(idxs []uint64) []uint64 {
+	indicesCount := len(idxs)
+	parents := make([]uint64, indicesCount)
+	for i := 0; i < indicesCount; i++ {
+		parents[i] = parentIndex(idxs[i])
+	}
+	parents = removeDuplicateIndices(parents)
```
Comment (blasrodri) [inline merkle/index.go:22]: if speed is what matters, you could keep a set of ids that are present, and avoid inserting the ones that match.

File: merkle/index.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+// siblingIndex returns index of a sibling element
+func siblingIndex(idx uint64) uint64 {
+	if isLeftIndex(idx) {
+		// Right sibling index
+		return idx + 1
+	}
+	// Left sibling index
+	return idx - 1
```
Comment (blasrodri) [inline merkle/index.go:33]: I think you can replace all this by using XOR
```suggestion
	return idx ^ 1
```

File: merkle/index.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+func getRightIndex(idx int) int {
+	return getLeftIndex(idx) + 1
+}
+
+// parentIndex returns index of a parent element
+func parentIndex(idx uint64) uint64 {
+	if isLeftIndex(idx) {
+		return idx / 2
+	}
+	return siblingIndex(idx) / 2
```
Comment (blasrodri) [inline merkle/index.go:53]:
```suggestion
	return (idx ^ 1) / 2
```

File: merkle/index.go (hunk 4 above)
Comment (vjdmhd) [reply merkle/index.go:6]: This does not going to map anything, it returns slice of all sibling indices

File: merkle/common_test.go (hunk 1 above)
Comment (vjdmhd) [reply merkle/common_test.go]: fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)

File: merkle/common_test.go (hunk 2 above)
Comment (vjdmhd) [reply merkle/common_test.go:79]: fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)

File: merkle/index.go (hunk 3 above)
Comment (vjdmhd) [reply merkle/index.go]: fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)

File: merkle/index.go (hunk 5 above)
Comment (vjdmhd) [reply merkle/index.go]: fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)

File: merkle/index.go (hunk 6 above)
Comment (vjdmhd) [reply merkle/index.go]: fixed in [this](https://github.com/ComposableFi/go-merkle-trees/pull/22/commits/2e360cdbbfc6c3e12c3c206b0c3b3d92e91c702c)

File: merkle/index.go (hunk 4 above)
Comment (blasrodri) [reply merkle/index.go:4]: Ok, thanks!

**Review by blasrodri — APPROVED** [review]

**Pull request merged**